From a89543f0737b296cf3a621371bdc3f49cf53afd6 Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Fri, 30 Jan 2015 22:30:46 +0800
Subject: [PATCH] meta-galileo, meta-iot-devkit: fix Arduino launcher fail
 issue in SPI image

The change on galileod.sh is to make sure that Arduinio launcher takes into
consideration of Galileo Gen 2 board.

The change on linux-yocto_3.8.bbappend is to make sure that both SD Card &
SPI image are using g_serial which is associated with linux-cd-acm driver.
This is the driver being distributed as part of Arduino IDE for Quark platform.

Reported-and-tested-by: Dermot Murphy <dermot.b.murphy@intel.com>
Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 .../recipes-core/initscripts/files/galileod.sh     |    9 ++++++++-
 .../linux/linux-yocto-quark_3.8.bbappend           |    4 ++--
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/meta-galileo/recipes-core/initscripts/files/galileod.sh b/meta-galileo/recipes-core/initscripts/files/galileod.sh
index 7a25528..447efe3 100755
--- a/meta-galileo/recipes-core/initscripts/files/galileod.sh
+++ b/meta-galileo/recipes-core/initscripts/files/galileod.sh
@@ -9,7 +9,14 @@ start_handler()
 {
     type dmidecode > /dev/null 2>&1 || die "dmidecode not installed"
     board=$(dmidecode -s baseboard-product-name)
-    case "$board" in *"Galileo" ) start_galileod ;; esac
+    case "$board" in
+        *"Galileo" )
+                   start_galileod
+                   ;;
+        *"GalileoGen2" )
+                   start_galileod
+                   ;;
+    esac
 }
 
 start_galileod()
diff --git a/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend b/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
index cf38728..b01c1a0 100644
--- a/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
+++ b/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
@@ -2,8 +2,8 @@ module_autoload_iwlwifi_iot-devkit = "iwlwifi"
 module_autoload_btusb = "btusb"
 
 # swap g_serial for g_acm_ms
-module_autoload_pch_udc = " pch_udc g_acm_ms"
-module_conf_g_acm_ms = "options g_acm_ms file=/dev/mmcblk0p1 removable=1 idVendor=0x8086 idProduct=0xDEAD"
+# module_autoload_pch_udc_iot-devkit = " pch_udc g_acm_ms"
+# module_conf_g_acm_ms_iot-devkit = "options g_acm_ms file=/dev/mmcblk0p1 removable=1 idVendor=0x8086 idProduct=0xDEAD"
 
 # PPP module autoload
 module_autoload_pppox = "pppox"
-- 
1.7.9.5

